From cb614dd430f864ec8b7e82c496aa2cc15a1c690e Mon Sep 17 00:00:00 2001
From: Valentine Burley <valentine.burley@collabora.com>
Date: Mon, 7 Sep 2026 14:36:59 +0200
Subject: [PATCH] vulkan/runtime: Export runtime dependencies in interface
 dependencies

idep_vulkan_runtime_body and idep_vulkan_lite_runtime link the
runtime static libraries but did not declare dependencies.

As a result, downstream consumers (such as driver shared libraries) did
not inherit external dependencies required by the runtime, most notably
idep_u_gralloc (and thus libui) on Android.

Signed-off-by: Valentine Burley <valentine.burley@collabora.com>
Reviewed-by: Yiwei Zhang <zzyiwei@chromium.org>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/44278>
---
 src/vulkan/runtime/meson.build | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/vulkan/runtime/meson.build b/src/vulkan/runtime/meson.build
index 64f95bd639f3..404f51662ad0 100644
--- a/src/vulkan/runtime/meson.build
+++ b/src/vulkan/runtime/meson.build
@@ -243,14 +243,14 @@ idep_vulkan_lite_runtime_headers = declare_dependency(
 if get_option('backend').startswith('vs')
   idep_vulkan_lite_runtime = declare_dependency(
     link_with : [libvulkan_lite_runtime, libvulkan_lite_instance],
-    dependencies : idep_vulkan_lite_runtime_headers
+    dependencies : [idep_vulkan_lite_runtime_headers, vulkan_lite_runtime_deps]
   )
 else
   idep_vulkan_lite_runtime = declare_dependency(
     # Instruct users of this library to link with --whole-archive.  Otherwise,
     # our weak function overloads may not resolve properly.
     link_whole : [libvulkan_lite_runtime, libvulkan_lite_instance],
-    dependencies : idep_vulkan_lite_runtime_headers
+    dependencies : [idep_vulkan_lite_runtime_headers, vulkan_lite_runtime_deps]
   )
 endif
 
@@ -318,10 +318,12 @@ libvulkan_instance = static_library(
 if get_option('backend').startswith('vs')
   idep_vulkan_runtime_body = declare_dependency(
     link_with : [libvulkan_lite_runtime, libvulkan_runtime, libvulkan_instance],
+    dependencies : vulkan_runtime_deps,
   )
 else
   idep_vulkan_runtime_body = declare_dependency(
     link_whole : [libvulkan_lite_runtime, libvulkan_runtime, libvulkan_instance],
+    dependencies : vulkan_runtime_deps,
   )
 endif
 
-- 
GitLab

